cli/internal/packagemanager/packagemanager_test.go

```diff
@@ -205,41 +205,43 @@ func Test_GetWorkspaces(t *testing.T) {
 		wantErr  bool
 	}
 
+	cwd, _ := os.Getwd()
+
 	rootPath := map[string]string{
-		"nodejs-npm":   "../../../examples/basic",
-		"nodejs-berry": "../../../examples/basic",
-		"nodejs-yarn":  "../../../examples/basic",
-		"nodejs-pnpm":  "../../../examples/with-pnpm",
+		"nodejs-npm":   filepath.Join(cwd, "../../../examples/basic"),
```

Comment:
This looks like it's changing what we're testing. Previously we were passing in relative paths, and now we're passing in absolute paths. Have we changed what the production code is doing?

Reply:
Previously `afero`'s `OsFs` implicitly relied on the current working directory. Eventually it delegates down to `Os.Open()` and it just kind of happens to work even though it's not supposed to be a valid path inside of an `fs.FS` filesystem (and happily broke everything when I swapped in a "valid" `fs.FS` implementation). In fact, the [leading path traversal is *also* not allowed for `fs.FS` filesystems](https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/io/fs/fs.go;l=47-72), which `afero.OsFs` claims to be.

We're lucky, however: the place that consumes this code is already set up to do the right thing:
https://github.com/vercel/turborepo/blob/5ad04a308d0d761e2bc2e1be76f16d6f7c4745f4/cli/internal/context/context.go#L156-L175

`workspaces` coming back fully-qualified immediately get relative-ized to make sure that they're children, and from that point forward the `workspaces` are no longer used.

So yes, in this case we *are* changing what we're testing, but to correct behavior, and I traced through the code to ensure that the consuming location didn't care. This change serves to lock in the current state of the world.

That this combination of edge cases and what amount to bugs in `afero` worked to begin with *astounds* me. Like playing Minesweeper with 999 mines and randomly clicking your way to success.